cloud/libvirt/actuators/machine/utils/domain.go

```diff
@@ -745,3 +754,125 @@ func DomainExists(machine *clusterv1.Machine) (bool, error) {
 
 	return true, nil
 }
+
+// LookupDomainByUUID retrieves a libvirt domain by UUID using the
+// provider config on the given machine.
+func LookupDomainByUUID(machine *clusterv1.Machine, uuid *uuid.UUID) (*libvirt.Domain, error) {
+	machineProviderConfig, err := MachineProviderConfigFromClusterAPIMachineSpec(&machine.Spec)
+	client, err := buildClient(machineProviderConfig.URI)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to build libvirt client: %s", err)
+	}
+
+	virConn := client.libvirt
+	if virConn == nil {
+		return nil, fmt.Errorf(LibVirtConIsNil)
+	}
+
+	dom, err := virConn.LookupDomainByUUID(uuid[:])
```

Comment:
Having got this far... I'm wondering why we pass a pointer to the uuid? Is it because it's an underlying C/native type?

Reply:
Ended up dropping the UUID thing altogether and just looking up by name. It's simpler and should be fine.